### ISSUE TITLE
SQLSTATE[HY000]: General error: 1832 Cannot change column 'user_id': used in a foreign key constraint 'FK_2AEFE01     7A76ED395'

### ISSUE THREAD

**Issue by Aerogirl10** [symfony/symfony#30828]
**Symfony version(s) affected**: x.y.z
4.2.3
**Description**  
<!-- A clear and concise description of the problem. -->
I am getting following error:
```
  An exception occurred while executing 'ALTER TABLE micro_post CHANGE user_id user_id INT NOT NULL':                
                                                                                                                     
  SQLSTATE[HY000]: General error: 1832 Cannot change column 'user_id': used in a foreign key constraint 'FK_2AEFE01  
  7A76ED395'  
```


LONG VERSION
```
-> ALTER TABLE micro_post CHANGE user_id user_id INT NOT NULL
Migration 20190402000548 failed during Execution. Error An exception occurred while executing 'ALTER TABLE micro_post CHANGE user_id user_id INT NOT NULL':

SQLSTATE[HY000]: General error: 1832 Cannot change column 'user_id': used in a foreign key constraint 'FK_2AEFE017A76ED395'

In AbstractMySQLDriver.php line 106:
                                                                                                                     
  An exception occurred while executing 'ALTER TABLE micro_post CHANGE user_id user_id INT NOT NULL':                
                                                                                                                     
  SQLSTATE[HY000]: General error: 1832 Cannot change column 'user_id': used in a foreign key constraint 'FK_2AEFE01  
  7A76ED395'                                                                                                         
                                                                                                                     

In PDOConnection.php line 90:
                                                                                                                     
  SQLSTATE[HY000]: General error: 1832 Cannot change column 'user_id': used in a foreign key constraint 'FK_2AEFE01  
  7A76ED395'                             
                                                                          
                    
In PDOConnection.php line 88:
                                                                                                                     
  SQLSTATE[HY000]: General error: 1832 Cannot change column 'user_id': used in a foreign key constraint 'FK_2AEFE01  
  7A76ED395'        
```  

It happens when I add 
`* @ORM\JoinColumn(nullable=false)`
in Post Entity and try to make a migration.

I have previous migrations in my folder and some data made with Fixtures in a database. **And** that seems to be the problem (?)
I've found info that the thing is that foreign key prevent from making changes, as they can make a mess in the structure (but doesn't check that)


**How to reproduce**  
<!-- Code and/or config needed to reproduce the problem. If it's a complex bug,
     create a "bug reproducer" as explained in:
     https://symfony.com/doc/current/contributing/code/reproducer.html -->

**Possible Solution**  
<!--- Optional: only if you have suggestions on a fix/reason for the bug -->
I had this problem a couple of times. Sometimes I got back to the point of making entity and then I made everything as I want it to be eventually... (but it creates other problems)
I think that once I just dropped the database, deleted all migrations and filled it again.

Unfortunately, none of those solutions will work on a big project and what's more important... Even if I make it work - I still don't really understand what is the problem and where this foreign key comes in (I thought I can somehow drop/delete this key in migrations files but there are just SQL statement so... I don't know what am I doing basically)

**Comment by Aerogirl10** [comment]
I think I've got it! 😄 Just add **ONE** silly thing...
`     * @ORM\JoinColumn(nullable=false, `**onDelete="CASCADE"**`)
`

Links that helped me:
[this](https://github.com/doctrine/data-fixtures/issues/261)
[this](https://stackoverflow.com/questions/8065279/deleting-a-foreign-key-in-symfony-doctrine)
[and this](https://github.com/doctrine/orm/issues/5665)

I must say that I saw it before but I didn't think that it may be the case... But at this point, I became so desperate that I copied every line I found on issues (if I knew where to paste them... lol) and finally something happened. 
It may be related to the question that makerBundle is asking when making relations

>  Do you want to automatically delete orphaned App\Entity\MicroPost objects (orphanRemoval)? (yes/no) [no]:
**Maybe there is a problem - I set nullable to be false but then I didn't want to remove orphaned elements?**

FYI: Before I add this line I again deleted and created relations between my entities with make **and there** I pointed that I want this column to be **not** nullable (previously, when makerBundle asked me if this field may be nullable I just left default "yes" and edited it manually) - migration still didn't work at this point so I deleted all migration files I had (ok, just moved them to temporary directory, I am not that YOLO) in project and DB. I don't think those actions made it work but if I misunderstood the whole process and onDelete="CASCADE" is doing something totally different - I just prefer to tell you my steps, maybe it will be useful for somebody.
Thanks to my mum, rock and roll and @dmaicher for giving my post confused emoji. Somehow it motivated me to try the easiest solutions - worked and now I am 😕 but 😊
